lib/web_ui/lib/src/ui/window.dart

```diff
@@ -638,6 +643,36 @@ abstract class Window {
   Brightness get platformBrightness => _platformBrightness;
   Brightness _platformBrightness = Brightness.light;
 
+  /// Reference to css media query that indicates the user theme preference on the web
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  /// Updates the platformBrightness with the new user preferenceq
+  ///
+  ///  The param brightnessMediaQuery can be a MediaQueryListEvent or a MediaQueryList;
+  ///
+  void _brightnessMediaQueryChanged(brightnessMediaQuery) {
+      Brightness previousPlatformBrightness = _platformBrightness;
+      _platformBrightness = brightnessMediaQuery.matches ? Brightness.dark : Brightness.light;
```

Comment:
in terms of browser support, it would be interesting to know what `brightnessMediaQuery.matches` returns when the browser doesn't support the `(prefers-color-scheme: dark)` query.

See: https://caniuse.com/#search=prefers-color-scheme

Reply:
It sets to false i think:
https://stackoverflow.com/a/10748541/8096916

[Here](https://www.w3.org/TR/mediaqueries-4/#evaluating) it says:
If the result of any of the above productions is used in any context that expects a two-valued boolean, “unknown” must be converted to “false”.
Note: This means that, for example, when a media query is used in a @media rule, if it resolves to “unknown” it’s treated as “false” and fails to match.

Reply:
Okey, I will move it to engine.window